Candidate entity and mail integration

ectomorph312 opened the pull request:
Wrote a dtoService to the candidate CRUD entity methods.
Integrated yandex mail, the first method receives all emails, then reads all emails.
Message consists of msgId, from, subject, text, fileName, size, receivedAt

ectomorph312 committed:
Candidate entity and mail integration

Wrote a dtoService to the candidate CRUD entity methods.
Integrated yandex mail, the first method receives all emails, then reads all emails.
Message consists of msgId, from, subject, text, fileName, size, received At

Farkhat commented on src/main/java/io/fortylines/hrcrm/dtoService/Impl/DefaultEmailDtoService.java:
```diff
@@ -0,0 +1,152 @@
+        properties.put("mail.smtp.host", "smtp.yandex.ru");
+        properties.put("mail.smtp.auth", true);
+        properties.put("mail.smtp.socketFactory.port", "465");
+        properties.put("mail.smtp.socketFactory.class", "javax.net.ssl.SSLSocketFactory");
```
Думаю эти значения стоит перенести в application.properties. Если они уже там, удалить строки

ectomorph312 replied:
переместил, но не удалил, а подставляю значения через аннотацию @Value

ectomorph312 committed:
Optimization of DefaultEmailServiceDto

Added methods setProperties( to avoid code duplication), getUnreadMessages, delete method(when calling the method, the findFileAndDelete method is called in the body, which searches for the file and deletes it)

ectomorph312 committed:
Optimization of DefaultEmailServiceDto and Added entity Feedback

Added methods setProperties( to avoid code duplication), getUnreadMessages, delete method(when calling the method, the findFileAndDelete method is called in the body, which searches for the file and deletes it)
Added entity Feedback + CRUD methods.
Implemented email distribution

ectomorph312 committed:
Update .gitignore

aidar4j commented on src/main/java/io/fortylines/hrcrm/controller/EmailController.java:
```diff
@@ -0,0 +1,53 @@
+    }
+
+    @DeleteMapping("/delete")
+    public void delete(@RequestParam Integer id, @RequestParam String fileName) throws MessagingException {
```
Why Integer instead of Long?

aidar4j replied:
Id should be sent via path variable

ectomorph312 replied:
changed via @PathVariable, I catch the file names in the same way, via @RequestParam

aidar4j commented on src/main/java/io/fortylines/hrcrm/controller/EmailController.java:
```diff
@@ -0,0 +1,53 @@
+        return emailDtoService.getUnreadMessages();
+    }
+
+    @DeleteMapping("/delete")
```
Deletemapping is already saying that it is a delete. Remove please /delete in path.

ectomorph312 replied:
deleted

aidar4j commented on src/main/java/io/fortylines/hrcrm/controller/EmailController.java:
```diff
@@ -0,0 +1,53 @@
+    }
+
+    @PostMapping("/mail-sending")
+    public void mailSending(@RequestParam String subject,
```
Use Dto instead of multiple params

ectomorph312 replied:
changed

aidar4j commented on src/main/java/io/fortylines/hrcrm/controller/EmailController.java:
```diff
@@ -0,0 +1,53 @@
+        emailDtoService.delete(id, fileName);
+    }
+
+    @PostMapping("/mail-sending")
```
Simple slash would be enough I think

ectomorph312 replied:
changed

aidar4j commented on src/main/java/io/fortylines/hrcrm/controller/EmailController.java:
```diff
@@ -0,0 +1,53 @@
+import java.util.List;
+
+@RestController
+@RequestMapping("api/v1/mail")
```
APIs are always in plural form

ectomorph312 replied:
changed

aidar4j commented on src/main/java/io/fortylines/hrcrm/controller/FeedbackController.java:
```diff
@@ -0,0 +1,44 @@
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequestMapping("api/v1/feedback")
```
APIs are always in plural form

ectomorph312 replied:
changed

aidar4j commented on src/main/java/io/fortylines/hrcrm/controller/FeedbackController.java:
```diff
@@ -0,0 +1,44 @@
+@RestController
+@RequestMapping("api/v1/feedback")
+@PreAuthorize("hasAnyRole('ROLE_ADMIN', 'ROLE_HR', 'ROLE_HEADOFDEPARTMENT')")
+public class FeedbackController {
```
Add method "getFeedbacksByCandidate" please

ectomorph312 replied:
added

aidar4j commented on src/main/java/io/fortylines/hrcrm/dtoService/Impl/DefaultCandidateDtoService.java:
```diff
@@ -0,0 +1,100 @@
+    }
+
+    @Override
+    public ReadCandidateDto create(MultipartFile resume ,CreateCandidateDto createCandidateDto) throws IOException {
```
Ctrl + Alt + L

ectomorph312 replied:
+

ectomorph312 replied:
changed

aidar4j commented on src/main/java/io/fortylines/hrcrm/dtoService/Impl/DefaultCandidateDtoService.java:
```diff
@@ -0,0 +1,100 @@
+    @Override
+    public ReadCandidateDto create(MultipartFile resume ,CreateCandidateDto createCandidateDto) throws IOException {
+        Candidate newCandidate = new Candidate();
+        Vacancy vacancyId = vacancyService.getById(createCandidateDto.getVacancyId());
```
vacancy, not vacancyId

ectomorph312 replied:
changed

aidar4j commented on src/main/java/io/fortylines/hrcrm/dtoService/Impl/DefaultCandidateDtoService.java:
```diff
@@ -0,0 +1,100 @@
+    public ReadCandidateDto update(Long id, MultipartFile resume,
+                                   UpdateCandidateDto updateCandidateDto) throws IOException {
+        Candidate updateCandidate = new Candidate();
+        Vacancy vacancyId = vacancyService.getById(updateCandidateDto.getVacancyId());
```
same

ectomorph312 replied:
changed

ectomorph312 committed:
Working on comments

removed extra padding
created a DTO for methods with multiple parameters
changing the names of variables
moved smtp data to application properties
